diff -ur bircd.old/src/s_bsd.c bircd/src/s_bsd.c
--- bircd.old/src/s_bsd.c	Sat Mar 26 10:27:18 2005
+++ bircd/src/s_bsd.c	Sat Apr  9 21:42:12 2005
@@ -1037,7 +1037,7 @@
   {
   get_sockhost(cptr,inetntoa6(&addr6.sin6_addr));
   bcopy((char *)&addr6.sin6_addr,(char *)cptr->ip,sizeof(struct in6_addr));
-  cptr->port=ntohs(addr6.sin_port);
+  cptr->port=ntohs(addr6.sin6_port);
   }
 else
 #endif
@@ -1457,7 +1457,7 @@
     {
     get_sockhost(acptr,inetntoa6(&addr6.sin6_addr));
     bcopy((char *)&addr6.sin6_addr,(char *)acptr->ip,sizeof(struct in6_addr));
-    acptr->port=ntohs(addr6.sin_port);
+    acptr->port=ntohs(addr6.sin6_port);
     }
   else
   #endif
diff -ur bircd.old/src/version.c bircd/src/version.c
--- bircd.old/src/version.c	Sat Mar 26 10:30:12 2005
+++ bircd/src/version.c	Sat Apr  9 21:41:00 2005
@@ -19,7 +19,7 @@
 
 #include "struct.h"
 
-#define VER "2.0.3rc11"
+#define VER "2.0.3rc11+fix1"
 
 char *creation = __DATE__ " " __TIME__;
 char *version = "bIRCd" VER;
