diff -ur bircd.orig/Config bircd/Config
--- bircd.orig/Config	2005-06-12 10:15:30.000000000 +0200
+++ bircd/Config	2005-07-28 14:52:04.891190416 +0200
@@ -46,12 +46,12 @@
 DEFLIBS="none"
 OSNAME="an unrecgonized operating system"
 #
-DPATH="/usr/local/etc/ircd"
-SPATH="/usr/local/bin/ircd"
-BPATH="/usr/local/bin"
-SET_UID=1
-SET_GID=1
-HUB=""
+DPATH="/etc/ircd"
+SPATH="/usr/sbin/ircd"
+BPATH="/usr/sbin"
+SET_UID=75
+SET_GID=75
+HUB="Yes"
 CRYPT_OPER_PASS=""
 CRYPT_SALT1="0xefcdab89"
 CRYPT_SALT2="0x98badcfe"
@@ -61,7 +61,7 @@
 NICKNAMEHISTORYLENGTH="2000"
 UNIXPORT=""
 USE_POLL=""
-MAXCONNECTIONS="128"
+MAXCONNECTIONS="512"
 FD_SETSIZE=""
 
 #
@@ -79,8 +79,6 @@
   n='-n'
 fi
 
-clear
-
 if [ "$1" = "-m" ]; then
   AUTO_CONFIG=""
 fi
@@ -898,16 +896,18 @@
 echo "#endif" >> $SETUP_H
 
 FOO=""
+runonce=""
 while [ -z "$FOO" ] ; do
-    if [ "$USER" = "root" ]; then
-      FOO="$DPATH"
-    else
-      FOO="${HOME}/ircd"
-    fi
+    FOO="$DPATH"
     echo ""
     echo "What directory are all the server configuration files in?"
     echo $n "[$FOO] -> $c"
-    read cc
+    if [ -z "$AUTO_CONFIG" -o -n "$runonce" ] ; then
+      read cc
+    else
+      cc=""
+      runonce=Yes
+    fi
     if [ -z "$cc" ] ; then
       cc=$FOO
     fi
@@ -918,16 +918,18 @@
 rm -f Makefile.tmp
 
 FOO=""
+runonce=""
 while [ -z "$FOO" ] ; do
-    if [ "$USER" = "root" ]; then
-      FOO="$BPATH"
-    else
-      FOO="${HOME}/ircd"
-    fi
+    FOO="$BPATH"
     echo ""
     echo "What directory is the ircd binary file in?"
     echo $n "[$FOO] -> $c"
-    read cc
+    if [ -z "$AUTO_CONFIG" -o -n "$runonce" ] ; then
+      read cc
+    else
+      cc=""
+      runonce=Yes
+    fi
     if [ -z "$cc" ] ; then
       cc=$FOO
     fi
@@ -976,7 +978,12 @@
     echo ""
     echo "Are you running as a HUB?"
     echo $n "[$FOO] -> $c"
-    read cc
+    if [ -z "$AUTO_CONFIG" -o -n "$runonce" ] ; then
+      read cc
+    else
+      cc=""
+      runonce=Yes
+    fi
     if [ -z "$cc" ] ; then
       cc=$FOO
     fi
@@ -1007,7 +1014,12 @@
     echo ""
     echo "Encrypt O-Line password in ircd.conf?"
     echo $n "[$FOO] -> $c"
-    read cc
+    if [ -z "$AUTO_CONFIG" -o -n "$runonce" ] ; then
+      read cc
+    else
+      cc=""
+      runonce=Yes
+    fi
     if [ -z "$cc" ] ; then
       cc=$FOO
     fi
@@ -1199,7 +1211,12 @@
     echo "If you aren't sure, answer NO. select() will be used instead."
     echo "Use poll() for non-blocking I/O?"
     echo $n "[$FOO] -> $c"
+    if [ -z "$AUTO_CONFIG" -o -n "$runonce" ] ; then
       read cc
+    else
+      cc=""
+      runonce=Yes
+    fi
     if [ -z "$cc" ] ; then
       cc=$FOO
     fi
@@ -1227,7 +1244,12 @@
     echo "How many file descriptors (or sockets) can the irc server use?"
     echo "(This value should be the multiply of 64)"
     echo $n "[$FOO] -> $c"
-    read cc
+    if [ -z "$AUTO_CONFIG" -o -n "$runonce" ] ; then
+      read cc
+    else
+      cc=""
+      runonce=Yes
+    fi
     if [ -z "$cc" ] ; then
       cc=$FOO
     fi
@@ -1363,10 +1385,8 @@
 #define NICKNAMEHISTORYLENGTH $NICKNAMEHISTORYLENGTH
 __EOF__
 
-if [ "$USER" = "root" ]; then
-  echo "#define SET_UID $SET_UID" >> $OPTIONS_H
-  echo "#define SET_GID $SET_GID" >> $OPTIONS_H
-fi
+echo "#define SET_UID $SET_UID" >> $OPTIONS_H
+echo "#define SET_GID $SET_GID" >> $OPTIONS_H
 
 if [ -n "$USE_POLL" ]; then
   echo "#define USE_POLL" >> $OPTIONS_H
diff -ur bircd.orig/doc/example.conf bircd/doc/example.conf
--- bircd.orig/doc/example.conf	2004-02-21 17:45:02.000000000 +0100
+++ bircd/doc/example.conf	2005-07-28 14:49:40.331166880 +0200
@@ -13,13 +13,16 @@
 # I line says who is allowed to connect, this one allows everyone
 I:*::::10
 
+# IPv6 addresses may be used in this file like in example below
+#I:[2001:1234::/48]::::10
+
 # IRCop line type /oper <nick> <Pass> to get power
-O:*@192.168.0.4:pass:admin:ao:8
+O:*@127.0.0.1:pass:admin:ao:8
 
 # N/C/H Lines are what you use when you link.
-N:192.168.0.1:pass:irc.hub::110
-C:192.168.0.1:pass:irc.hub:7325:110
-H:*::irc.hub
+#N:192.168.0.1:pass:irc.hub::110
+#C:192.168.0.1:pass:irc.hub:7325:110
+#H:*::irc.hub
 
 # P lines are the port lines these say what ports can be used
 P:*:::6667
diff -ur bircd.orig/include/config.h bircd/include/config.h
--- bircd.orig/include/config.h	2005-06-12 09:44:34.000000000 +0200
+++ bircd/include/config.h	2005-07-28 14:49:40.333166576 +0200
@@ -37,7 +37,7 @@
 **
 ** Define this to enable IPv6 support.
 */
-//#define INET6
+#define INET6
 
 /* NO_DEFAULT_INVISIBLE
 **
@@ -77,7 +77,7 @@
 ** Define it to have extra logging to users.log, opers.log and rejects.log
 ** Function is fuzzy and I dont recommend to use it. -- Borg
 */
-//#define LOGGING
+#define LOGGING
 
 /* NO_DNS
 **
@@ -221,15 +221,15 @@
  * You must define these to something, even if you don't really want them.
  */
 #ifndef DPATH
-#define DPATH   "."             /* dir where all ircd stuff is */
+#define DPATH   "/var/lib/ircd"             /* dir where all ircd stuff is */
 #endif
 #ifndef SPATH
-#define SPATH   "."             /* path to server executeable */
+#define SPATH   "/usr/sbin/ircd"             /* path to server executeable */
 #endif
-#define CPATH   "ircd.conf"     /* server configuration file */
-#define MPATH   "ircd.motd"     /* server MOTD file */
-#define LPATH   "debug.log"     /* Where the debug file lives, if DEBUGMODE */
-#define PPATH   "ircd.pid"      /* file for server pid */
+#define CPATH   "/etc/ircd/ircd.conf"     /* server configuration file */
+#define MPATH   "/etc/ircd/ircd.motd"     /* server MOTD file */
+#define LPATH   "/var/log/ircd/debug.log"     /* Where the debug file lives, if DEBUGMODE */
+#define PPATH   "/var/lib/ircd/ircd.pid"      /* file for server pid */
 
 /*
  * Define this filename to maintain a list of persons who log
@@ -239,9 +239,9 @@
  * successful use of /oper. FNAME_CONNLOG logs all rejected connections.
  * These are either full paths or files within DPATH.
  */
-#define FNAME_USERLOG "users.log"
-#define FNAME_OPERLOG "opers.log"
-#define FNAME_CONNLOG "rejects.log"
+#define FNAME_USERLOG "/var/log/ircd/users.log"
+#define FNAME_OPERLOG "/var/log/ircd/opers.log"
+#define FNAME_CONNLOG "/var/log/ircd/rejects.log"
 
 
 /* CHROOTDIR
