diff -ur bircd.orig/Makefile.dist bircd/Makefile.dist
--- bircd.orig/Makefile.dist	2003-05-03 17:15:14.000000000 +0200
+++ bircd/Makefile.dist	2004-08-02 18:19:52.000000000 +0200
@@ -77,6 +77,7 @@
 #
 DPATH=
 BPATH=
+DESTDIR=
 
 # [CHANGEME]
 # Some SunOS versions want this.  Try it without first.
@@ -153,4 +154,4 @@
 	@cd src; ${MAKE} ${MAKEARGS} depend
 
 install:
-	@./ircdinst $(BPATH) $(DPATH)
+	@./ircdinst $(DESTDIR)/$(BPATH) $(DESTDIR)/$(DPATH)
